frontend/packages/dev-console/src/components/import/devfile/DevfileStrategySection.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import { Alert, TextInputTypes } from '@patternfly/react-core';
+import { FormikValues, useFormikContext } from 'formik';
+import { useTranslation } from 'react-i18next';
+import { getGitService } from '@console/git-service/src';
+import { InputField } from '@console/shared/src';
+import { safeYAMLToJS } from '@console/shared/src/utils/yaml';
+import FormSection from '../section/FormSection';
+import { useDevfileServer } from './devfileHooks';
+import DevfileInfo from './DevfileInfo';
+
+const DevfileStrategySection: React.FC = () => {
+  const { t } = useTranslation();
+  const { values, setFieldValue } = useFormikContext<FormikValues>();
+  const {
+    git: { url, type, ref, dir, secretResource },
+    devfile,
+  } = values;
+  const [, devfileParseError] = useDevfileServer(values, setFieldValue);
+
+  const getDevfileInfo = () => {
+    const devfileJSON = safeYAMLToJS(values.devfile.devfileContent);
+    return {
+      displayName: devfileJSON.metadata?.name || 'Devfile',
+      tags: devfileJSON.metadata?.version ? [devfileJSON.metadata.version] : [],
+      iconClass: devfileJSON.metadata?.name ? `icon-${devfileJSON.metadata?.name}` : '',
+    };
+  };
+
+  const handleDevfileChange = React.useCallback(async () => {
+    const gitService = getGitService(url, type, ref, dir, secretResource, devfile.devfilePath);
+    if (!values.devfile?.devfileSourceUrl) {
+      // No need to check the existence of the file, waste of a call to the gitService for this need
+      const devfileContents = gitService && (await gitService.getDevfileContent());
```

Comment:
will it be safe here to put await in try-catch block?

Reply:
@invincibleJai the getDevfileContent util already catches fetch errors and returns null if any.

Reply:
We dont need try-catch here as git-service already handling this.